src/node_crypto.cc

```diff
@@ -3320,6 +3320,9 @@ void Hmac::HmacDigest(const FunctionCallbackInfo<Value>& args) {
 
   bool r = hmac->HmacDigest(&md_value, &md_len);
   if (!r) {
+    if ( md_value != nullptr ) {
+        delete[] md_value;
+    }
```

Comment:
The code is a little confusing though.  Replacing the body of the `if` block with `CHECK_EQ(nullptr, md_value); CHECK_EQ(0, md_len)` would make it clearer that no memory allocation is expected to happen on failure.